controllers/front/DiscountController.php

```diff
@@ -43,30 +43,53 @@ public function initContent()
         $nb_cart_rules = count($cart_rules);
 
         foreach ($cart_rules as $key => &$discount ) {
+
             if ($discount['quantity_for_user'] === 0) {
                 unset($cart_rules[$key]);
             }
 
-
             $discount['value'] = Tools::convertPriceFull(
-                                            $discount['value'],
-                                            new Currency((int)$discount['reduction_currency']),
-                                            new Currency((int)$this->context->cart->id_currency)
-                                        );
+                $discount['value'],
+                new Currency((int)$discount['reduction_currency']),
+                new Currency((int)$this->context->cart->id_currency)
+            );
+            
             if ($discount['gift_product'] !== 0) {
-                $product = new Product((int) $discount['gift_product']);
-                if (isset($product->name)) {
-                    $discount['gift_product_name'] = current($product->name);
+                $product = new Product((int) $discount['gift_product'], false, (int)$this->context->language->id);
+                if (!Validate::isLoadedObject($product) || !$product->isAssociatedToShop() || !$product->active) {
+                    unset($cart_rules[$key]);
                 }
+                if (Combination::isFeatureActive() && $discount['gift_product_attribute'] !== 0) {
+                    $attributes = $product->getAttributeCombinationsById((int)$discount['gift_product_attribute'], (int)$this->context->language->id);
+                    $gift_attributes = array();
```

Comment:
Whenever you create a new variable, please use the camelCase notation (use our [coding standards](http://doc.prestashop.com/display/PS16/PHP+Coding+Standards) for reference).